mlflow/types/schema.py

```diff
@@ -630,28 +672,72 @@ def from_json_dict(cls, **kwargs):
             return SparkMLVector()
         if kwargs["values"]["type"] == MAP_TYPE:
             return cls(value_type=Map.from_json_dict(**kwargs["values"]))
+        if kwargs["values"]["type"] == ANY_TYPE:
+            return cls(value_type=AnyType())
         return cls(value_type=kwargs["values"]["type"])
 
-    def _merge(self, map_type: "Map") -> "Map":
-        if not isinstance(map_type, Map):
-            raise MlflowException(f"Can't merge map with non-map type: {type(map_type).__name__}")
-        if self == map_type:
+    def _merge(self, other: BaseType) -> Map:
+        if isinstance(other, AnyType) or self == other:
             return deepcopy(self)
+        if not isinstance(other, Map):
+            raise MlflowException(f"Can't merge map with non-map type: {type(other).__name__}")
         if isinstance(self.value_type, DataType):
-            if self.value_type == map_type.value_type:
+            if self.value_type == other.value_type:
                 return Map(value_type=self.value_type)
             raise MlflowException(
                 f"Map types are incompatible for {self} with value_type={self.value_type} and "
-                f"{map_type} with value_type={map_type.value_type}"
+                f"{other} with value_type={other.value_type}"
             )
 
-        if (
-            isinstance(self.value_type, (Array, Object, Map))
-            and self.value_type.__class__ is map_type.value_type.__class__
-        ):
-            return Map(value_type=self.value_type._merge(map_type.value_type))
+        if isinstance(self.value_type, (Array, Object, Map, AnyType)):
+            return Map(value_type=self.value_type._merge(other.value_type))
+
+        raise MlflowException(f"Map type {self!r} and {other!r} are incompatible")
+
+
+@experimental
+class AnyType:
+    def __init__(self):
+        """
+        AnyType can store any json-serializable data including None values.
+        For example:
+
+        .. code-block::python
+
+            from mlflow.types.schema import AnyType, Schema, ColSpec
+
+            schema = Schema([ColSpec(type=AnyType(), name="id")])
 
-        raise MlflowException(f"Map type {self!r} and {map_type!r} are incompatible")
+        .. Note::
+            AnyType should be used when the field is None, the type is not known
+            at the time of data creation, or the field can have multiple types.
+            e.g. for GenAI flavors, the model output could contain `None` values,
+            and `AnyType` can be used to represent them.
+            AnyType has no data validation at all, please be aware of this when
+            using it.
+        """
+
+    def __repr__(self) -> str:
+        return "Any"
+
+    def __eq__(self, other) -> bool:
+        return isinstance(other, AnyType)
+
+    def to_dict(self):
+        return {"type": ANY_TYPE}
+
+    def _merge(self, another_type: BaseType) -> BaseType:
```

Comment:
another_type -> other.

Reply:
Addressed comments in the follow up PR https://github.com/mlflow/mlflow/pull/13846